test-utils-core/src/main/java/uk/gov/justice/services/test/utils/core/schema/SchemaComplianceChecker.java

```diff
@@ -0,0 +1,371 @@
+                        .sorted()
+                        .collect(Collectors.toList());
+
+        if (duplicatedSchemasList.isEmpty()) {
+            this.isCompliant = true;
+            this.numberOfViolatingSchemas = 0;
+            return;
+        }
+
+        final Map<String, Set<String>> offenders = new HashMap<>();
+
+        duplicatedSchemasList.forEach(e -> {
+            Set<String> setOfDistinctVersions = schemaFilesList
+                    .stream()
+                    .filter(schemaEntry -> schemaEntry.getFilename().equals(e))
+                    .map(SchemaEntry::getChecksum)
+                    .distinct()
+                    .collect(Collectors.toSet());
+            if (setOfDistinctVersions.size() > 1) {
+                LOG.error("{} has {} distinct versions", e, setOfDistinctVersions.size());
+            } else {
+                LOG.debug("{} is the same across modules", e);
+            }
+            if (setOfDistinctVersions.size() > 1) {
+                offenders.put(e, setOfDistinctVersions);
+            }
+        });
+
+        final Map<String, String> schemaContentHashMap =
+                offenders
+                        .entrySet()
+                        .stream()
+                        .map(e -> e.getValue())
+                        .flatMap(l -> l.stream())
+                        .collect(Collectors.toSet()).stream().collect(Collectors.toMap(e -> e, Function.identity()));
+
+        schemaContentHashMap
+                .entrySet()
+                .stream()
+                .forEach(eSet -> {
+                    final URL url = schemaFilesList
+                            .stream()
+                            .filter(e -> e.getChecksum().equals(eSet.getValue()))
+                            .map(SchemaEntry::getUrl)
+                            .findAny()
+                            .get();
+
+                    String fileContents = null;
+                    fileContents = readURL(url);
+                    eSet.setValue(fileContents);
+                });
+
+
+        if (offenders.size() < 1) {
+            this.isCompliant = true;
+            LOG.info("Schema compliance passed. {} files checked in total", this.totalSchemasChecked);
+        } else {
+
+            this.numberOfViolatingSchemas = offenders.size();
+            this.isCompliant = false;
+            final String report = generateReport(schemaFilesList, offenders, schemaContentHashMap);
+            LOG.error(report);
+        }
+
+    }
+
+    private void computeChecksums(List<SchemaEntry> schemaFilesList) {
+        schemaFilesList
+                .forEach((SchemaEntry e) ->
+                {
+                    e.setChecksum(calculateSHA256Checksum(e.getUrl()));
+
+                    LOG.debug("{} - {}", e.getFilename(), e.getChecksum());
+                });
+    }
+
+    private List<SchemaEntry> readAllFilesFromDirs(Enumeration<URL> schemaDir) throws SchemaComplianceException {
```

Comment:
Almost half of the code in this class is about reading files, consider move those into a separate class

Reply:
well the main class is really about reading files. there's one section which is about checking what's different

Reply:
The class is called SchemaComplianceChecker. It's taking on two distinct responsibilities of reading schemas and validating schemas. One of the responsibilities should be offloaded.